src/EditorFeatures/CSharpTest/IntroduceUsingStatement/IntroduceUsingStatementTests.cs

```diff
@@ -676,6 +676,401 @@ void M()
 
         var d = 1;
     }
+}");
+        }
+
+        [Fact, WorkItem(43001, "https://github.com/dotnet/roslyn/issues/43001")]
+        public async Task ConsumeFollowingTryStatement1()
+        {
+            await TestInRegularAndScriptAsync(
+@"using System.IO;
+
+class C
+{
+    void M()
+    {
+        var reader = new MemoryStream()[||];
+        try
+        {
+            var buffer = reader.GetBuffer();
+            int a = buffer[0], b = a;
+            var c = b;
+            var d = 1;
+        }
+        finally
+        {
+            reader.Dispose();
+        }
+    }
+}",
+@"using System.IO;
+
+class C
+{
+    void M()
+    {
+        using (var reader = new MemoryStream())
+        {
+            var buffer = reader.GetBuffer();
+            int a = buffer[0], b = a;
+            var c = b;
+            var d = 1;
+        }
+    }
+}");
+        }
+
+        [Fact, WorkItem(43001, "https://github.com/dotnet/roslyn/issues/43001")]
+        public async Task ConsumeFollowingTryStatement2()
+        {
+            await TestInRegularAndScriptAsync(
+@"using System;
+using System.IO;
+
+class C
+{
+    void M()
+    {
+        var reader = new MemoryStream()[||];
+        try
+        {
+            var buffer = reader.GetBuffer();
+            int a = buffer[0], b = a;
+            var c = b;
+            var d = 1;
+        }
+        catch (Exception e)
+        {
+        }
+        finally
+        {
+            reader.Dispose();
+        }
+    }
+}",
+@"using System;
+using System.IO;
+
+class C
+{
+    void M()
+    {
+        using (var reader = new MemoryStream())
+        {
+            try
+            {
+                var buffer = reader.GetBuffer();
+                int a = buffer[0], b = a;
+                var c = b;
+                var d = 1;
+            }
+            catch (Exception e)
+            {
+            }
+            finally
```

Comment:
in this case can we not get rid of the finally (but keep the catch?)

Reply:
er........ yes.  good point.  I think i will choose to not support that until someone asks :)